eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Gov.hs

```diff
@@ -440,7 +440,7 @@ legalProtVer ::
   Maybe (GovActionId (EraCrypto era), (ProtVer, ProtVer))
 legalProtVer pp pgaids ps gas = do
   (newProtVer, prevProtVer) <- preceedingHardFork (gasAction gas) pp pgaids ps
-  if pvCanFollow newProtVer (SJust prevProtVer)
+  if pvCanFollow prevProtVer newProtVer
```

Comment:
Can I suggest we maybe flip the order of the arguments in `pvCanFollow`? Right now it reads really weirdly, "prevProtVer can follow newProtVer". That's a recipe for disaster..?

Reply:
I do agree with you that it reads weirdly, but unfortunately we should not change the order now because that would be dangerous. Not only because this function has been around for a long time and we don't know if there are other users of it out there, but also because this is function is part of the Shelley spec. So we would have to amend the specification as well. 

All in all it is not worth it. Good documentation and clearer implementation should be enough for users to not create a disaster.

Reply:
how about we deprecate `pvCanFollow` and introduce `canBumpTo` which has the arguments the right way around?

Reply:
Or we can keep the function around for our implementation but not export it and instead export only `canBumpTo`.

Reply:
Order of arguments is not important! As long as documentation clearly states what function does that is the most important part.

It's like arguing on the order of buffers in `memcpy`, it is all a matter of perspective.

Reply:
Argument order clearly matters, as evidenced by the bug..?

Reply:
It matters that the arguments are passed in correctly, but the interpretation of their order is subjective. I think changing the order would cause problems than solve.

Reply:
> Argument order clearly matters, as evidenced by the bug..?

@MaximilianAlgehed Lack of documentation is the cause of the bug, not the order of the arguments!

Which everyone should learn from: writing documentation for function is important!

I am guilty of this as well, so if you ever see me forgetting haddock in my PRs, especially for exported functions please call me out on it.